src/Features/CSharp/Portable/UseExpressionBodyForLambda/UseExpressionBodyForLambdaCodeStyleProvider.cs

```diff
@@ -0,0 +1,436 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseExpressionBodyForLambda
+{
+    internal class UseExpressionBodyForLambdaCodeStyleProvider
+        : AbstractCodeStyleProvider<ExpressionBodyPreference, SyntaxKind, UseExpressionBodyForLambdaCodeStyleProvider>
+    {
+        private static readonly LocalizableString UseExpressionBodyTitle = new LocalizableResourceString(nameof(FeaturesResources.Use_expression_body_for_lambda_expressions), FeaturesResources.ResourceManager, typeof(FeaturesResources));
+        private static readonly LocalizableString UseBlockBodyTitle = new LocalizableResourceString(nameof(FeaturesResources.Use_block_body_for_lambda_expressions), FeaturesResources.ResourceManager, typeof(FeaturesResources));
+
+        public UseExpressionBodyForLambdaCodeStyleProvider() 
+            : base(CSharpCodeStyleOptions.PreferExpressionBodiedLambdaExpressions,
+                   IDEDiagnosticIds.UseExpressionBodyForLambdaExpressionsDiagnosticId,
+                   UseExpressionBodyTitle,
+                   UseExpressionBodyTitle)
+        {
+        }
+
+        #region analysis
```

Comment:
note: in practice, the desired pattern might be to place these regions in separate parts.